x/programs/rust/sdk-macros/src/lib.rs

```diff
@@ -267,12 +267,9 @@ pub fn state_keys(_attr: TokenStream, item: TokenStream) -> TokenStream {
     const MAX_VARIANTS: usize = u8::MAX as usize + 1;
 
     if variants.len() > MAX_VARIANTS {
-        return Error::new(
-            variants[MAX_VARIANTS].span(),
-            "Cannot exceed `u8::MAX` variants",
-        )
-        .into_compile_error()
-        .into();
+        return Error::new(variants[MAX_VARIANTS].span(), "Cannot exceed 256 variants")
```

Comment:
nit: i'd rather not hardcode 256, and say `u8::MAX + 1` or even better 

`format!("Cannot exceed {} variants", MAX_VARIANTS))`

 but i'm not too opinionated on this.

Reply:
Thanks! Applied your suggestion to use `MAX_VARIANTS`, it's better indeed.